pkg/controller.v1/common/job.go

```diff
@@ -307,9 +307,15 @@ func (jc *JobController) ReconcileJobs(
 				MinResources:      minResources,
 			}
 
-			_, err := jc.SyncPodGroup(metaObject, pgSpec)
+			pg, err := jc.SyncPodGroup(metaObject, pgSpec)
 			if err != nil {
 				log.Warnf("Sync PodGroup %v: %v", jobKey, err)
+				return nil
+			}
+
+			if pg == nil || pg.Status.Phase == "" || pg.Status.Phase == v1beta1.PodGroupPending {
+				log.Warnf("PodGroup %v unschedulable", jobKey)
+				return nil
```

Comment:
Did we watch PodGroup?

I am not sure if there is a new reconciliation loop for the status change.

Reply:
seems not, maybe we need to update the job status here to trigger a new reconciliation loop, l will give it a try.